File: hw/tipan/u3/hw17/VideoGameSystemJava/src/main/java/utils/MongoManager.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package utils;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;

public class MongoManager {

    private static final String URI = "mongodb+srv://Ronald:Ronald@cluster0.cd2ybxo.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0";

    public static MongoDatabase getDatabase() {

        MongoClient mongoClient = MongoClients.create(URI);

        return mongoClient.getDatabase("VideoGameDB");
    }
}

File: hw/chavez/u3/hw21AnimalFarm/javascript/utils/MongoManager.js
const { MongoClient } = require("mongodb");

class MongoManager {

    constructor() {
        this.uri = "mongodb+srv://odalys:odalys@cluster0.2cf9puv.mongodb.net/AnimalFarmDB?retryWrites=true&w=majority&appName=Cluster0";
        this.client = new MongoClient(this.uri);
        this.database = null;
    }

    async connect() {
        await this.client.connect();
        this.database = this.client.db("AnimalFarmDB");
        console.log("Connected to MongoDB Atlas");
        return this.database;
    }

    async close() {
        await this.client.close();
        console.log("Connection closed");
    }

}

module.exports = MongoManager;